### PR TITLE
Fix documentation generation

### PR DESCRIPTION
**Pull request by gnunicorn**
We see moving bindings docs into the exising directory fails often on CI. This creates a new folder and moves all the different docs into that before publishing this folder instead of the previous one.

### CODE REVIEW

**Review by jplatte — CHANGES_REQUESTED** [review]
This creates two directories, `doc` and `docs`.

**Comment by gnunicorn** [comment]
damn... see, this is why you need to take breaks! Your brain gets foggy, you see what you want to see and make stupid mistakes like these...

File: .github/workflows/documentation.yml [hunk 1]
```diff
@@ -56,15 +56,17 @@ jobs:
       - name: Prepare the doc hierarchy
         shell: bash
         run: |
-          mkdir -p target/docs/bindings/matrix-sdk-crypto-nodejs/
-          mkdir -p target/docs/bindings/matrix-sdk-crypto-js/
-          mv -f bindings/matrix-sdk-crypto-nodejs/docs/* target/docs/bindings/matrix-sdk-crypto-nodejs/
-          mv -f bindings/matrix-sdk-crypto-js/docs/* target/docs/bindings/matrix-sdk-crypto-js/
+          mkdir -p doc
```
Comment (jplatte) [inline .github/workflows/documentation.yml:59]: Unnecessary, created as part of the next line (because of `-p`).

```suggestion
```

**Review by jplatte — APPROVED** [review]
I don't really understand how this fix works, but it doesn't look *wrong*.

**Comment by gnunicorn** [comment]
the problem is that in the existing code, the cache puts back `target/doc` so when we create docs, we can't just _move_ the JS into it (because some already exists). In this patch, we put all together into a separate `doc`-folder, which isn't in the cache, and we move all in there fresh, before uploading it all to github pages.

**Pull request merged**